### DISCUSSION TITLE
New project, how to configure typescript?

### DISCUSSION THREAD

**Discussion by DeanKamali** [directus/directus#6663] [Archive]
Hey everyone, 

this may seem to be a trivia question, just can't figure out how to configure typescript with a new directus project. 

can someone guide me by sharing your `package.json` and `tsconfig.json `? 

Thanks

**Comment by rijkvanzanten** [comment]
Are you talking about extensions development specifically? Or just how to initialize your own project that uses Directus as a back-end API?

**Comment by DeanKamali** [comment]
Hi @rijkvanzanten 

I'm talking about using typescript as an extension for development. 

For example, when using API Hooks, would be nice to get IntelliSense to work with knexjs. 

I couldn't find an option to enable it when initializing a new project via  `npx create-directus-project my-project`

**Reply by rijkvanzanten** [reply]
`npx create-directus-project my-project` just scaffolds a new Directus project, but doesn't setup any extensions for you. The new extensions sdk that @nickrum already has a TS boilerplate to initialize IIRC

**Reply by nickrum** [reply]
Typescript extension scaffolding is not available yet as it requires a bit more refactoring, but it will be very soon.

**Reply by DeanKamali** [reply]
@nickrum looking forward to it, thank you

**Reply by dstoyanoff** [reply]
Hey @nickrum, is there an issue for it so we can track it? I am also interested in typescript extensions

**Reply by nickrum** [reply]
@dstoyanoff there is already a pull request: #6881
